app/services/broadcasts/welcome_notification/generator.rb

```diff
@@ -11,27 +10,49 @@ def self.call(*args)
       end
 
       def call
-        return if commented_on_welcome_thread? || received_notification?
+        return unless user
+
+        send_welcome_notification
+        send_authentication_notification
+      end
+
+      def send_welcome_notification
+        return if received_notification?(welcome_broadcast) || commented_on_welcome_thread?
 
         Notification.send_welcome_notification(user.id, welcome_broadcast.id)
       end
 
-      def received_notification?
-        Notification.exists?(notifiable: welcome_broadcast, user: user)
+      def send_authentication_notification
+        return if received_notification?(authentication_broadcast) || authenticated_with_both_services? || user.created_at > 1.day.ago
+
+        Notification.send_welcome_notification(user.id, authentication_broadcast.id)
+      end
+
+      private
+
+      attr_reader :user
+
+      def received_notification?(broadcast)
+        Notification.exists?(notifiable: broadcast, user: user)
       end
 
       def commented_on_welcome_thread?
         welcome_thread = Article.admin_published_with("welcome").first
         Comment.where(commentable: welcome_thread, user: user).any?
       end
 
-      private
+      def authenticated_with_both_services?
+        user.identities.exists?(provider: "github") && user.identities.exists?(provider: "twitter")
```

Comment:
nice! i wouldn't have easily figured out about `user.identities` :)

Reply:
This can be simplified by using a single query instead of two:

```ruby
user.identities.where(provider: %w[github twitter]).count == 2
```

Reply:
Updated it to @rhymes 's suggestion